website/src/pages/docs/features/explicit-resource-management.mdx

```diff
@@ -0,0 +1,210 @@
+your resources. This is especially important when you are dealing with resources that need to be
+cleaned up when they are no longer needed, or clean up the operations in a queue when the server is
+shutting down.
+
+## Dispose GraphQL Yoga
+
+GraphQL Yoga supports
+[Explicit Resource Management](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html#using-declarations-and-explicit-resource-management)
+approach that allows you to dispose of resources when they are no longer needed. This can be done in
+two ways.
+
+<Tabs items={['`dispose` method', '`await using` syntax']}>
+  <Tabs.Tab>
+    ```ts
+    console.log('Yoga is starting')
+    {
+        await using yoga = createYoga({
+            schema: createSchema({
+                typeDefs: /* GraphQL */ `
+                    type Query {
+                        hello: String
+                    }
+                `,
+                resolvers: {
+                    Query: {
+                        hello: () => 'world'
+                    }
+                }
+            })
+        });
+    }
+    console.log('Yoga is disposed')
+    ```
+    </Tabs.Tab>
+    <Tabs.Tab>
+    ```ts
+    console.log('Yoga is starting')
+    const yoga = createYoga({
+        schema: createSchema({
+            typeDefs: /* GraphQL */ `
+                type Query {
+                    hello: String
+                }
+            `,
+            resolvers: {
+                Query: {
+                    hello: () => 'world'
+                }
+            }
+        })
+    });
+    await yoga.dispose();
+    console.log('Yoga is disposed')
+    ```
+    </Tabs.Tab>
+</Tabs>
+
+In the first example, we use the `await using` syntax to create a new instance of `yoga` and dispose
+of it when the block is exited. In the second example, we create a new instance of `yoga` and
+dispose of it using the `dispose` method.
+
+### Dispose on Node.js
+
+When running GraphQL Yoga on Node.js, you can use process event listeners or server's `close` event
+to trigger GraphQL Yoga's disposal. Or you can configure GraphQL Yoga to handle this automatically
+by listening `process` exit signals.
+
+<Tabs items={['Explicit disposal', 'Automatic disposal on process exit']}>
+
+<Tabs.Tab>
+```ts
+import { createYoga, createSchema } from 'graphql-yoga';
+import { createServer } from 'http';
+
+const yoga = createYoga({ schema: createSchema({ typeDefs: /_ GraphQL _/
+`             type Query {                 hello: String             }         `, resolvers: {
+Query: { hello: () => 'world' } } }) });
+
+const server = createServer(yoga); server.listen(4000, () => { console.info('Server is running on
+http://localhost:4000/graphql') }); server.once('close', async () => { await yoga.dispose();
+console.info('Server is disposed so is Yoga') });
+
+// Then when the server is shutting down, it will dispose of Yoga server.close();
+
+````
```

Comment:
this code is broken by prettier try to add new line before ```ts and after ```

Reply:
Thanks @dimaMachina !